### PR TITLE
Interop profile

### PR DESCRIPTION
**Pull request by tulshi**
Incorporates suggestions from Apoorva and Shayne based on discussions in past WG meetings.

### CODE REVIEW

File: caep-interoperability-profile-1_0.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+: Receivers MUST interpret all allowable values of this field. Transmitters MAY generate any allowable value of this field
+
+`credential_type`
+: Receivers MUST interpret the following values of this field. Transmitters MAY generate any of the following values:
+
+  * `password`
+  * `pin`
+  * `fido2-platform`
+  * `fido2-roaming`
+  * `fido2-u2f`
```
Comment (appsdesh) [inline caep-interoperability-profile-1_0.md:187]: We should include all the existing values that exist in the CAEP spec

File: caep-interoperability-profile-1_0.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+### Implicitly Added Subjects {#common-receiver-subjects}
+Receivers MUST assume that all subjects are implicitly included in a Stream, without any `AddSubject` method invocations.
+
+## Event Subjects {#common-event-subjects}
+The following subject identifier formats from "Subject Identifiers for Security Event Tokens" {{RFC9493}} MUST be supported:
+
+* `email`
+* `iss_sub`
+
+Receivers MUST be prepared to accept events with any of the subject identifier formats specified in this section. Transmitters MUST be able to send events with at least one of subject identifier formats specified in this section.
```
Comment (appsdesh) [inline caep-interoperability-profile-1_0.md:162]: Can we rephrase this as "at least one" for the receiver as well, otherwise it adds more burden on the receiver than the transmitter

File: caep-interoperability-profile-1_0.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+: Receivers MUST be able to create a Stream with the Transmitter using valid authorization with the Transmitter. The Transmitter MAY support multiple streams with the same Receiver
+
+**Reading Stream Configuration**
+: A Receiver MUST be able to obtain current Stream configuration from the Transmitter by providing a valid authorization
+
+**Getting the Stream Status**
+: A Receiver MUST be able to obtain the current Stream status from the Transmitter by providing a valid authorization
+
+**Stream Verification**
+: A Receiver MUST be able to verify the liveness of the Stream by requesting that the Transmitter send it a Stream Verificaiton event
```
Comment (appsdesh) [inline caep-interoperability-profile-1_0.md:145]: This is missing term "valid authorization" like other APIs

File: caep-interoperability-profile-1_0.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+The following requirements are common across all use-cases defined in this document.
+
+## Transmitters {#common-transmitters}
+Transmitters MUST implement the following features:
+
+### Spec Version {#spec-version}
+The Transmitter Configuration Metadata MUST have a `spec_version` field, and its value MUST be `1_0-ID2` or greater
+
+### Delivery Method {#delivery-method}
+The Transmitter Configuration Metadata MUST include the `delivery_methods_supported` field.
```
Comment (appsdesh) [inline caep-interoperability-profile-1_0.md:97]: Let's call out PUSH is a MUST ( this will align with restrictions on the `delivery` object mentioned below)

File: caep-interoperability-profile-1_0.md (hunk 1 above)
Comment (tulshi) [reply caep-interoperability-profile-1_0.md:187]: Since it is a "MUST" include, I feel we can add a few more like "phone-voice" and "phone-sms" can be added, but I don't want to require Transmitters and Receivers to have to support all of them.

File: caep-interoperability-profile-1_0.md (hunk 4 above)
Comment (tulshi) [reply caep-interoperability-profile-1_0.md:97]: This was a [discussion](https://hackmd.io/@oidf-wg-sse/wg-meeting-20231128#Push-Support) we had a few weeks ago, because I had it earlier in the spec. I think folks said that requiring Receivers to implement PUSH will make compliance harder.

File: caep-interoperability-profile-1_0.md (hunk 2 above)
Comment (tulshi) [reply caep-interoperability-profile-1_0.md:162]: Having "at least one" on both sides can lead to incompatibilities, so for interoperability we can require both sides to support both subject types or only one side to support "at least one"

File: openid-caep-interoperability-profile-1_0.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+### Streams {#common-stream-configuration}
+In all streams created by the Transmitter, the following MUST be true:
+
+#### Delivery {#common-delivery}
+The `delivery` field MUST be present in the Configuration of any Stream generated by the Transmitter, and its value MUST include the following:
+
+~~~ json
+{
+    "method" : "urn:ietf:rfc:8935"
```
Comment (appsdesh) [inline openid-caep-interoperability-profile-1_0.md:128]: @atultulshi in your earlier comment you mentioned that PUSH is not a MUST have, then why do we suggest push to be a delivery method here?

File: openid-caep-interoperability-profile-1_0.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+      -
+        ins: A. Deshpande
+        name: Apoorva Deshpande
+        org: Okta
+        email: apoorva.deshpande@okta.com
+
+normative:
+  RFC9493: # Subject Identifier Formats for SETs
+  SSF:
+    target: http://openid.net/specs/openid-sse-framework-1_0.html
```
Comment (FragLegs) [inline openid-caep-interoperability-profile-1_0.md:35]: Are we able to link to draft 2, since this interop requires draft 2?

File: openid-caep-interoperability-profile-1_0.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+  RFC8935:
+
+
+--- abstract
+This document defines an interoperability profile for implementations of the Shared Signals Framework (SSF) {{SSF}} and the Continuous Access Evaluation Profile (CAEP) {{CAEP}}. It is organized around use-cases that improve security of authenticated sessions. It specifies certain optional elements from within the SSF and CAEP specifications as being required to be supported in order to be considered as an interoperable implementation.
+
+--- middle
+
+# Introduction {#introduction}
+SSF and CAEP together enable improved session security outcomes. This specification defines the minimum required features from SSF and CAEP that an implementation MUST offer in order to be considered as an interoperable implementation. This document defines specific use cases. An implementation may support only a subset of the use cases defined herein, and SHALL be considered an interoperable implementation for the specific use-cases it supports. The following use-cases are considered as a part of this specification:
```
Comment (FragLegs) [inline openid-caep-interoperability-profile-1_0.md:79]:
```suggestion
SSF and CAEP together enable improved session security outcomes. This specification defines the minimum required features from SSF and CAEP that an implementation MUST offer in order to be considered as an interoperable implementation. This document defines specific use cases. An implementation MAY support only a subset of the use cases defined herein, and SHALL be considered an interoperable implementation for the specific use-cases it supports. The following use-cases are considered as a part of this specification:
```

File: openid-caep-interoperability-profile-1_0.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+
+Session Revocation
+: A SSF Transmitter or Receiver is able to respectively generate or respond to the CAEP session-revoked event
```
Comment (FragLegs) [inline openid-caep-interoperability-profile-1_0.md:82]:
```suggestion
: An SSF Transmitter or Receiver is able to respectively generate or respond to the CAEP session-revoked event
```

File: openid-caep-interoperability-profile-1_0.md [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+
+Credential Change
+: A SSF Transmitter or Receiver is able to respectively generate or respond to the CAEP credential-change event
```
Comment (FragLegs) [inline openid-caep-interoperability-profile-1_0.md:85]:
```suggestion
: An SSF Transmitter or Receiver is able to respectively generate or respond to the CAEP credential-change event
```

File: openid-caep-interoperability-profile-1_0.md [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+The Transmitter Configuration Metadata MUST have a `spec_version` field, and its value MUST be `1_0-ID2` or greater
+
+### Delivery Method {#delivery-method}
+The Transmitter Configuration Metadata MUST include the `delivery_methods_supported` field.
+
+### JWKS URI {#jwks-uri}
+The Transmitter Configuration Metadata MUST include the `jwks_uri` field, and its value MUST provide the current signing key of the Transmitter.
+
+### Configuration Endpoint {#configuration-endpoint}
+The Transmitter Configuration Metadata MUST include the `configuration_endpoint` field. The specified endpoint MUST provide a way to Create a Stream.
```
Comment (FragLegs) [inline openid-caep-interoperability-profile-1_0.md:103]:
```suggestion
The Transmitter Configuration Metadata MUST include the `configuration_endpoint` field. The specified endpoint MUST provide a way to create a stream.
```

File: openid-caep-interoperability-profile-1_0.md [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+
+### Status Endpoint {#status-endpoint}
+The Transmitter Configuration Metadata MUST include the `status_endpoint` field. The specified endpoint MUST provide a way to Get and Update the Stream Status. The Transmitter MUST be able to pause and restart streams. For streams that are paused, the Transmitter MUST specify (offline) the resource constraints on how many events it can keep, or for how long. The way a Transmitter specifies this information is outside the scope of the SSF spec.
```
Comment (FragLegs) [inline openid-caep-interoperability-profile-1_0.md:106]:
```suggestion
The Transmitter Configuration Metadata MUST include the `status_endpoint` field. The specified endpoint MUST provide a way to get and update the stream status. The Transmitter MUST be able to pause and restart streams. For streams that are paused, the Transmitter MUST specify (offline) the resource constraints on how many events it can keep, or for how long. The way a Transmitter specifies this information is outside the scope of the SSF spec.
```

File: openid-caep-interoperability-profile-1_0.md (hunk 5 above)
Comment (FragLegs) [reply openid-caep-interoperability-profile-1_0.md:128]: I agree. PUSH delivery gets set by the Receiver, not the Transmitter. So we should either say:
1. The Transmitter MUST include rfc:8936 in the delivery field
2. The Transmitter must be ready to accept a delivery specification of rfc:8935 from a Recevier

I actually think we should require both of those things.

File: caep-interoperability-profile-1_0.md [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+
+### Streams {#common-stream-configuration}
+In all streams created by the Transmitter, the following MUST be true:
+
+#### Delivery {#common-delivery}
+The `delivery` field MUST be present in the Configuration of any Stream generated by the Transmitter, and its value MUST include the following:
+
+~~~ json
+{
+    "method" : "urn:ietf:rfc:8935"
```
Comment (appsdesh) [inline caep-interoperability-profile-1_0.md:128]: Re-adding the comment from yesterday as it disappeared. 

@atultulshi could you please clarify why are we mandating 8935 here when we are going to flexible around PUSH and POLL

File: openid-caep-interoperability-profile-1_0.md (hunk 8 above)
Comment (tulshi) [reply openid-caep-interoperability-profile-1_0.md]: I'm generally using "a SSF" because it's "a Shared Signals Framework"... I think both can be argued to be correct.

File: openid-caep-interoperability-profile-1_0.md (hunk 5 above)
Comment (tulshi) [reply openid-caep-interoperability-profile-1_0.md]: A stream can only be one of "push" or "poll", so I think we cannot have the language @FragLegs suggests. I'll make it so that the Transmitter must be ready to accept either.

**Review by FragLegs — APPROVED** [review]

File: openid-caep-interoperability-profile-1_0.md (hunk 5 above)
Comment (FragLegs) [reply openid-caep-interoperability-profile-1_0.md]: Excellent. The change you made is what I was trying to suggest. Sorry for my unclear language.

**Review by appsdesh — APPROVED** [review]

File: caep-interoperability-profile-1_0.md (hunk 1 above)
Comment (timcappalli) [reply caep-interoperability-profile-1_0.md]: As a general comment, I think it is challenging to only support a subset of an event type. In my opinion, the most specific requirement should be the event type holistically, not certain parts of it.

A transmitter should only be expected to support the properties that is supports in its respective platform. Since we're not telling receivers what to do with a given property, they should only be required to validate that the property is present and of the correct format.

File: openid-caep-interoperability-profile-1_0.md [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+## Event Subjects {#common-event-subjects}
+The following subject identifier formats from "Subject Identifiers for Security Event Tokens" {{RFC9493}} MUST be supported:
+
+* `email`
+* `iss_sub`
+
+Receivers MUST be prepared to accept events with any of the subject identifier formats specified in this section. Transmitters MUST be able to send events with at least one of subject identifier formats specified in this section.
+
+## Event Signatures
+All events MUST be signed using the `RS256` algorithm using a minimum of 2048-bit keys.
```
Comment (FragLegs) [inline openid-caep-interoperability-profile-1_0.md:176]: I missed this in my first pass. What would you think of using `ES256` instead of (or in addition to) `RS256`? I recognize that `RS256` is often the default implementation, but `ES256` is more secure and generally preferred over `RS256`.

File: openid-caep-interoperability-profile-1_0.md [hunk 14]
```diff
@@ -0,0 +1,196 @@
...
+## Receivers {#common-receivers}
+Receivers MUST implement the following features:
+
+### Delivery Methods {#common-receiver-delivery}
+Receivers MUST be able to accept events using the Push-Based Security Event Token (SET) Delivery Using HTTP {{RFC8935}} specification and the Poll-Based Security Event Token (SET) Delivery Using HTTP {{RFC8936}} specification.
+
+### Implicitly Added Subjects {#common-receiver-subjects}
+Receivers MUST assume that all subjects are implicitly included in a Stream, without any `AddSubject` method invocations.
+
+## Event Subjects {#common-event-subjects}
```
Comment (FragLegs) [inline openid-caep-interoperability-profile-1_0.md:167]: Another question that just occurred to me - the SSF spec requires that Verification event subjects be given in the `opaque` format. Do we need to add `opaque` to this list?

**Review by FragLegs — CHANGES_REQUESTED** [review]

**Review by FragLegs — APPROVED** [review]
Approving so we can merge and keep working on it after

**Review by timcappalli — APPROVED** [review]
